### PR TITLE
Add a unique error message when workloads are disabled but a customer is trying to use workloads

### PR DESCRIPTION
**Pull request by marcpopMSFT**
Fixes https://github.com/dotnet/sdk/issues/33395

Not sure how common this will be or if it's worth it but it'll prevent some confusion and was easy enough to implement.

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]
Looks good.

File: src/Tasks/Common/Resources/Strings.resx [hunk 1]
```diff
@@ -899,4 +899,8 @@ You may need to build the project on another operating system or architecture, o
     <value>NETSDK1205: The Microsoft.Net.Compilers.Toolset.Framework package should not be set directly. Set the property 'BuildWithNetFrameworkHostedCompiler' to 'true' instead if you need it.</value>
     <comment>{StrBegin="NETSDK1205: "}{Locked="Microsoft.Net.Compilers.Toolset.Framework"}{Locked="BuildWithNetFrameworkHostedCompiler"}</comment>
   </data>
+  <data name="UnsupportedTargetPlatformIdentifierWithWorkloadsDisabled" xml:space="preserve">
+    <value>NETSDK1208: The target platform identifier {0} was not recognized. Please ensure that MSBuildEnableWorkloadResolver is set to true and .NET workloads are enabled</value>
```
[the remark this answers is not in the document]
Comment (marcpopMSFT) [reply src/Tasks/Common/Resources/Strings.resx]: Went with something similar to your suggestion though a little different.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets [hunk 2]
```diff
@@ -107,11 +107,16 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <PropertyGroup>
       <TargetPlatformSupported Condition="'$(TargetPlatformIdentifier)' == 'Windows'">true</TargetPlatformSupported>
+      <UseWorkloadsSpecificError Condition="($(TargetPlatformIdentifier.StartsWith('ios')) or $(TargetPlatformIdentifier.StartsWith('tvos')) or $(TargetPlatformIdentifier.StartsWith('maccatalyst')) or $(TargetPlatformIdentifier.StartsWith('android')) or $(TargetPlatformIdentifier.StartsWith('browser'))) and '$(MSBuildEnableWorkloadResolver)' != 'true'">true</UseWorkloadsSpecificError>
```
[the remark this answers is not in the document]
Comment (marcpopMSFT) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets]: Changed them all to equals except for browser as I'm pretty sure I'd seen browser-wasm before as the TPI.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets [hunk 3]
```diff
@@ -107,11 +107,16 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <PropertyGroup>
       <TargetPlatformSupported Condition="'$(TargetPlatformIdentifier)' == 'Windows'">true</TargetPlatformSupported>
+      <UseWorkloadsSpecificError Condition="($(TargetPlatformIdentifier) == 'ios' or $(TargetPlatformIdentifier) == 'tvos' or $(TargetPlatformIdentifier) == 'maccatalyst' or $(TargetPlatformIdentifier) == 'android' or $(TargetPlatformIdentifier.StartsWith('browser'))) and '$(MSBuildEnableWorkloadResolver)' != 'true'">true</UseWorkloadsSpecificError>
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:110]: Does that mean `net8.0-browser` is ready to be used? @marcpopMSFT

Comment (marcpopMSFT) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:110]: I'll defer to @danroth27 I do see it listed in our documentation: https://learn.microsoft.com/en-us/dotnet/standard/frameworks#net-5-os-specific-tfms

Comment (danroth27) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:110]: @lewing Didn't we disable `net8.0-browser` before shipping .NET 8?

Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:110]: @danroth27 Apparently, no.

And as it's now in the official documentation, people are starting to use it. So, I assume it won't be disabled/removed in .NET 9?

**Pull request merged**